encoding/jsonbq/marshal.go

```diff
@@ -0,0 +1,47 @@
+package jsonbq
+
+import (
+	"cloud.google.com/go/bigquery"
+	"encoding/json"
+)
+
+const (
+	ModeRepeated string = "REPEATED"
+	ModeRequired string = "REQUIRED"
+	ModeNullable string = "NULLABLE"
+)
+
+// Field describes the schema of a field in BigQuery.
+type Field struct {
+	Name        string   `json:"name"`
+	Type        string   `json:"type"`
+	Mode        string   `json:"mode"`
+	Description string   `json:"description,omitempty"`
+	Fields      []*Field `json:"fields,omitempty"`
+}
+
+func MarshalSchema(schema bigquery.Schema) ([]byte, error) {
+	fields := convertSchemaToMarshableFields(schema)
+	return json.Marshal(fields)
+}
+
+func convertSchemaToMarshableFields(schema bigquery.Schema) []*Field {
+	fields := make([]*Field, len(schema))
+	for i, fieldSchema := range schema {
+		var field Field
+		field.Name = fieldSchema.Name
+		field.Type = string(fieldSchema.Type)
+		switch {
+		case fieldSchema.Repeated:
+			field.Mode = ModeRepeated
+		case fieldSchema.Required:
+			field.Mode = ModeRequired
+		default:
```

Comment:
return an error if we encounter an unhandled case here?

Reply:
The mode can only be one of these and it should always default to `NULLABLE`. [See this](https://cloud.google.com/bigquery/docs/schemas#modes). So I don't think it makes any sense to return an error here.

Reply:
This was originally an if-ifelse-else statement that the linter complained on.

Reply:
If it's an enum I think we should error on unexpected values rather than silently treat all unexpected values as singular

Reply:
We don't actually switch on any value, thus there is no unexpected value, really... But maybe I'm missing what you mean?